### ISSUE TITLE
stop second derivatives by stop_grad=True

### ISSUE THREAD

**Issue by csyanbin** [cbfinn/maml#8]
I noticed the new version of MAML paper said second derivatives has little influence on performance. 
So I run the code and using --stop_grad=True and all other setting are as default. 
The results seems 2% lower than --stop_grad=False.

My training cmd is as follows:
`python main.py --datasource=miniimagenet --metatrain_iterations=60000 --meta_batch_size=4 --update_batch_size=1 --update_lr=0.01 --num_updates=5 --num_classes=5 --logdir=logs_scope/miniimagenet1shot/ --num_filters=32 --max_pool=True --stop_grad=True`

Do I miss anything to stop second derivatives?

**Comment by Soyabean1994** [comment]
Do you solve this problem? I try to recover the MAML with pytorch without the second derivatives. My results for the miniimagenet is about 2% lower than results in the paper. I guess that the second derivatives makes the difference?

**Comment by csyanbin** [comment]
I cannot repeat the results without second derivatives. Maybe some implementation difference or training tricks? @cbfinn

**Comment by cbfinn** [comment]
Some potential ideas:
- I used early stopping, picking the model with the best validation accuracy during training, which wasn't necessarily the last one.
- The random seed on my computer was probably different than the one on your computer. [I only ran it once. Perhaps it had a particularly good set of initial weights / mini-batch ordering.]
- I don't think that I did any gradient clipping for the paper, but you could try clipping the meta-gradients.

**Comment by csyanbin** [comment]
Thanks for your explanation.

- For early stopping, I will try it and check the accuracy again.
- As to random seed, I think if we all use the same OS and same python version (2.7 for me) with same seed number, the random sequence will be the same. I follow the random seed in your released code.
- I notice there is gradient clipping in https://github.com/cbfinn/maml/blob/master/maml.py#L146 for miniImagenet dataset. 

According to your suggestions, I will try this experiment again without secondary derivatives.
